nixos/modules/services/misc/mautrix-whatsapp.nix

```
@@ -0,0 +1,131 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+  dataDir = "/var/lib/mautrix-whatsapp";
+  format = pkgs.formats.json {};
+
+  registrationFile = "${dataDir}/whatsapp-registration.yaml";
+  settingsFile = format.generate "config.json" cfg.settings;
+
+  startupScript = ''
+    ${pkgs.yq}/bin/yq -s '.[0].appservice.as_token = .[1].as_token
+      | .[0].appservice.hs_token = .[1].hs_token
+      | .[0]' ${settingsFile} ${registrationFile} \
+      > ${dataDir}/config.yml
+
+    ${pkgs.mautrix-whatsapp}/bin/mautrix-whatsapp \
+      --config='${dataDir}/config.yml' \
+      --registration='${registrationFile}'
+  '';
+in
+{
+  options.services.mautrix-whatsapp = {
+    enable = mkEnableOption "Mautrix-whatsapp, a puppeting bridge between Matrix and WhatsApp.";
+
+    settings = mkOption rec {
+      apply = recursiveUpdate default;
+      inherit (format) type;
+
+      description = ''
+        This options will be transform in YAML configuration file for the bridge
+
+        Look <link xlink:href="https://github.com/tulir/mautrix-whatsapp/wiki/Bridge-setup">here</link> for documentation.
+      '';
+      default = {
+        homeserver = {
+          domain = config.services.matrix-synapse.server_name;
+        };
+        appservice = rec {
+          address = http://localhost:29318;
+          hostname = "0.0.0.0";
+          port = 29318;
+          database = {
+            type = "sqlite3";
+            uri = "${dataDir}/mautrix-whatsapp.db";
+          };
+          id = "whatsapp";
+          bot = {
+            username = "whatsappbot";
+            displayname = "WhatsApp Bot";
+          };
+          as_token = "";
+          hs_token = "";
+        };
+        bridge = {
+          username_template = "whatsapp_{{.}}";
+          displayname_template = "{{if .Notify}}{{.Notify}}{{else}}{{.Jid}}{{end}}";
+          command_prefix = "!wa";
+          permissions."*" = "relaybot";
+        };
+        relaybot = {
```

```suggestion
          permissions."*" = "relay";
        };
        relay = {
```

This seems to have been changed to `relay` see [this commit](https://github.com/mautrix/whatsapp/commit/bb9a0f6528a0c5d2e81b3860da4ff58da33a9c52#diff-8b9ef9c277ec2256b84aab2ff2c3b778753ac96389c892c95847057365b66016)

---

`relay` should also be inside `bridge`. This also raises a bigger problem: invalid configurations are silently ignored.